pkg/cover/report.go

```diff
@@ -159,13 +159,19 @@ func (rg *ReportGenerator) generate(w io.Writer, progs []Prog, files map[string]
 		Root: new(templateDir),
 	}
 	for fname, file := range files {
-		if !strings.HasPrefix(fname, rg.buildDir) {
+		var remain string
+		if !filepath.IsAbs(fname) {
+			// If it's a relative path, it is relative to the build directory.
+			fname = filepath.Join(rg.buildDir, fname)
+			remain = filepath.Clean(strings.TrimPrefix(fname, rg.srcDir))
+		} else if !strings.HasPrefix(fname, rg.buildDir) {
```

Comment:
I wonder if we could only change the if condition in the orignal code to:
```
if filepath.IsAbs(fname) && !strings.HasPrefix(fname, rg.buildDir) {
```
So that we don't error on relative path. The rest may just work. We will try to strip buildDir, but it can't be a prefix (one starts with /, another doesn't), so that will be a no-op.

Reply:
I'm not sure how other OSes handle the relative path case. For fuchsia it is now special-cased.